src/components/authorization/authorization.service.ts

```diff
@@ -111,6 +114,41 @@ export class AuthorizationService {
     return parseSecuredProperties(props, permissions, resource.SecuredProps);
   }
 
+  async verifyCanEditChanges<TResource extends ResourceShape<any>>(
+    resource: TResource,
+    baseNode: Partial<TResource['prototype']>, // for use cases like checking defined files separately
```

Comment:
Why partial here?

Reply:
I guess at the time I wanted to be very clear that these are for the "simple" properties for the object and changes. I'm fine with taking out partial here, though.